test/test_listen_socket.cpp

```diff
@@ -0,0 +1,205 @@
+/*
+
+Copyright (c) 2016, Steven Siloti
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+* Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in
+the documentation and/or other materials provided with the distribution.
+* Neither the name of the author nor the names of its
+contributors may be used to endorse or promote products derived
+from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+#if TORRENT_USE_IPV6
+
+#include "test.hpp"
+#include "libtorrent/aux_/session_impl.hpp"
+
+using namespace libtorrent;
+
+namespace
+{
+	void test_equal(listen_socket_t const& s, address addr, int port, std::string dev, bool ssl)
+	{
+		TEST_EQUAL(s.ssl, ssl);
+		TEST_EQUAL(s.local_endpoint.address(), addr);
+		TEST_EQUAL(s.original_port, port);
+		TEST_EQUAL(s.device, dev);
+	}
+
+	void test_equal(aux::listen_endpoint_t const& e1, address addr, int port, std::string dev, bool ssl)
+	{
+		TEST_EQUAL(e1.ssl, ssl);
+		TEST_EQUAL(e1.port, port);
+		TEST_EQUAL(e1.addr, addr);
+		TEST_EQUAL(e1.device, dev);
+	}
+}
+
+TORRENT_TEST(partition_listen_sockets)
+{
+	{
```

Comment:
would you mind turning these blocks into individual test cases? I don't think they're coupled in any way, right?